### PR TITLE
Upgrade React-Query to V4

### PR DESCRIPTION
**Pull request by Zeko369**
Closes: #3543, #3291

### What are the changes and their implications?

Upgrades from react-query 3 -> 4

This also fixed the tests from #3291

https://twitter.com/tannerlinsley/status/1549785943837515776

### CODE REVIEW

**Comment by StringKe** [comment]
I have [question](https://github.com/blitz-js/blitz/issues/3595) about useQuery , how can I test with the current pr ?

**Comment by Zeko369** [comment]
@StringKe  You could merge it locally and run it locally, or wait for this to get merged and test that way 🤔

File: integration-tests/qm/package.json [hunk 1]
```diff
@@ -18,7 +18,7 @@
     "prisma": "4.0.0",
     "react": "18.0.0",
     "react-dom": "18.0.0",
-    "react-query": "3.39.0"
+    "react-query": "4.0.0-beta.23"
```
Comment (prisis) [inline integration-tests/qm/package.json:21]: Shouldn't this be changed to @tanstack/react-query ?

Comment (Zeko369) [reply integration-tests/qm/package.json:21]: Yeah, this was with the last react-query version (before official 4.0.0)

**Pull request merged**